### ISSUE TITLE
Configure Firebase witout Identity Platform

### ISSUE THREAD

**Issue by josh-burton** [hashicorp/terraform-provider-google#19774]
### Question

We've recently migrated a bunch of Firebase projects to Terraform, and have no realised that most of them now have Authentication with Identity Platform, rather than just standard Fireabse auth. There is a significant cost increase with identity platform.

What I can't see from the docs is when a project is configured with identity platform.

* Does the `google_identity_platform_config` resource always configure Identity Platform?
* Is it a timing issue? Should we be ensuring the `google_firebase_project` resource is configured first, before configuring the  `google_identity_platform_config` resource?
* Is the `identitytoolkit.googleapis.com` service required for standard Firebase Auth?

Appreciate any help or guidance!

**Comment by josh-burton** [comment]
Yes it is an optional upgrade via the UI, and should be via Terraform as well.

https://firebase.google.com/docs/projects/terraform/get-started#resources-authentication

These docs mention the following:

```
[google_identity_platform_config — enable Google Cloud Identity Platform (GCIP) (which is the backend for Firebase Authentication) and provide project-level authentication settings

Configuring Firebase Authentication via Terraform requires enabling GCIP.
```

**Comment by serpro69** [comment]
Ah, I see what you mean now. Since we actually use GCIP in firebase , I haven't even thought to try to setup firebase auth w/o it. Yeah, it shouldn't be mandatory via terraform if it's an optional feature. A bit odd why they'd do it like this, if it's intentional. Maybe better to create a bug/feature request instead?

You could also try to set things up manually via UI (w/o enabling the GCIP) and then export stuff into terraform via gcloud (see https://cloud.google.com/docs/terraform/resource-management/export) and see what resources that will export. It might be that it's possible to set this up w/o GCIP, just not documented.

**Comment by josh-burton** [comment]
Thats a good tip about exporting the terraform config. I have also filed a feature request with Firebase and created a user voice request: https://firebase.uservoice.com/forums/948424-general/suggestions/48939287-allow-configuring-standard-firebase-authentication

**Comment by rainshen49** [comment]
Hey @josh-burton , to confirm your exploration:

1. Plain old Firebase Auth is only enabled via the UI in Firebase Console
2. Terraform only supports the GCIP version (paid)

I was surprised as well initially. However, there were a variety of reasons behind this that made it difficult to support the old Firebase Auth in Terraform. Thank you for submitting the feature request to uservoice.
